lnd.go

```diff
@@ -385,11 +396,58 @@ func Main(cfg *Config, lisCfg ListenerCfg, interceptor signal.Interceptor) error
 	}
 	defer stopProxy()
 
-	// We wait until the user provides a password over RPC. In case lnd is
-	// started with the --noseedbackup flag, we use the default password
-	// for wallet encryption.
-	if !cfg.NoSeedBackup {
-		params, err := waitForWalletPassword(cfg, pwService, interceptor.ShutdownChannel())
+	// What wallet mode are we running in? We've already made sure the no
+	// seed backup and auto unlock aren't both set during config parsing.
+	switch {
+	// No seed backup means we're also using the default password.
+	case cfg.NoSeedBackup:
+		// We continue normally, the default password has already been
+		// set above.
+
+	// A password for unlocking is provided in a file.
+	case cfg.WalletUnlockPasswordFile != "":
+		ltndLog.Infof("Attempting automatic wallet unlock with " +
+			"password provided in file")
+		pwBytes, err := ioutil.ReadFile(cfg.WalletUnlockPasswordFile)
+		if err != nil {
+			return fmt.Errorf("error reading password from file "+
+				"%s: %v", cfg.WalletUnlockPasswordFile, err)
+		}
+
+		// Remove any newlines at the end of the file. The lndinit tool
+		// won't ever write a newline but maybe the file was provisioned
+		// by another process or user.
+		pwBytes = bytes.TrimRight(pwBytes, "\r\n")
+
+		// We have the password now, we can ask the unlocker service to
+		// do the unlock for us.
+		unlockedWallet, unloadWalletFn, err := pwService.LoadAndUnlock(
+			pwBytes, 0,
```

Comment:
do we want to have `recoveryWindow` passed in here as well? if so it seems like the remainder of the `default` clause (after line 461) could be shared between the two paths

Reply:
We don't have a `recoveryWindow` in this case as that's always sent through the unlocker RPC. I guess if they need to recover or rescan they'd need to remove the auto unlock flag and use the RPC instead.